### ISSUE TITLE
Using Photoswipe doesn't show up Alt and Title tag

### ISSUE THREAD

**Issue by djzwerg** [backdrop-contrib/gallery_tiler#5]
Hi,
thanks again for providing [Photoswipe support](https://github.com/backdrop-contrib/gallery_tiler/issues/1). As I recognized today the Alt and Title tag are not visible. Is it possible to provide these information within photoswipe?

**Comment by indigoxela** [comment]
> Is it possible to provide these information within photoswipe?

I have no idea, let me know, if you figured out. There's no obvious option, as far as I can see in the [official docs](https://photoswipe.com/options/).

**Comment by djzwerg** [comment]
Ok, I will have a look. If you choose Photoswipe as a field formatter in core or in Views it shows these tags.

**Comment by djzwerg** [comment]
Just tried a bit but I'm not able to figure it out. Gallery tiler shows alt and title but the colorbox or photoswipe modal doesn't. If you use one of them as a field formatter in core or views (without gallery tiler) they show the values too. It's a bit annoying but I don't know how to send the values from gallery tiler to the modal.

**Comment by indigoxela** [comment]
> If you use one of them as a field formatter in core or views (without gallery tiler) they show the values too

"Show" is pretty misleading here. If setting the field display to photoswipe, the _images in the overlay have no alt attribute_ at all, so you probably mean something completely else ... caption eventually?

You'd set the caption via "data-overlay-title".

**Comment by djzwerg** [comment]
Yes I mean that normally the title tag is used for caption and the alt tag by the screenreader.

Neither Photoswipe nor Colorbox are having captions when they are used as a modal by Gallery Tiler. The alt tags maybe can be ignored because Gallery Tiler Providers them before opening the images in a modal.

**Comment by indigoxela** [comment]
> Yes I mean that normally the title tag is used for caption

The "title tag" is no tag, but an HTML attribute to show the text on mouse-over. Modules like Colorbox or Photoswipe *may* use that attribute to show a caption instead. They also have a setting for that.

Gallery Tiler simply does not care about overlay captions. So if you need that, you might want to switch to a module that provides captions in overlays.

OR you could use hook_field_attach_view_alter() to adapt that yourself in a custom module/theme. Personally, I don't need the captions.

**Comment by indigoxela** [comment]
Oh, just had a quick idea, how you can add a caption via template...

Override the field-gallery-tiler.tpl.php, for example by adding it to your theme, then...

```
...
    <?php foreach ($items as $delta => $item): ?>
      <?php
        // Use the image alt as overlay caption:
        if (!empty($item['#item']['alt'])) {
          $item['#path']['options']['attributes']['data-overlay-title'] = $item['#item']['alt'];
        }
      ?>
...
```

**Comment by indigoxela** [comment]
I gave it a quick try (have to review again, when I have more time), but a pull request is available for testing: #6.

**Comment by djzwerg** [comment]
Thanks a lot @indigoxela for your great and fast work! This PR works for me - solving this feature request 😄
